metaflow/plugins/kfp/kfp_s3_sensor.py

```diff
@@ -46,22 +97,28 @@ def path_formatter_template(key: str, flow_parameters: dict) -> str:
     else:
         if os_expandvars:
             # expand OS env variables
-            path = os.path.expandvars(path)
+            path: str = os.path.expandvars(path)
         # default variable substitution
         path: str = path.format(**flow_parameters)
 
     # debugging print statement for customers so they know the final path
     # we're looking for
     print(f"Waiting for path: {path}...")
 
-    parsed_path = urlparse(path)
+    parsed_path: ParseResult = urlparse(path)
+    bucket: str
+    key: str
     bucket, key = parsed_path.netloc, parsed_path.path.lstrip("/")
 
-    s3 = boto3.client("s3")
-    start_time = time.time()
+    previous_elapsed_time: float = read_elapsed_time_s3_path(flow_name, kfp_run_id)
+
+    s3: botocore.client.BaseClient
+    s3_client_error: ClientError
+    s3, s3_client_error = get_s3_client()
```

Comment:
This s3 client is unused since you already put them in the read / write s3 client function. Meanwhile I assume recreating s3 client for each loop is less ideal. 

Can you modify the read and write function to take a passed in s3_client so that it is not recreated every loop?